cucumber-messages/dotnet/messages.proto

```diff
@@ -396,6 +399,26 @@ message PickleStepArgument {
   }
 }
 
+////// TestCases
+
+message TestCase {
```

Comment:
Is this a new message or where is it used? Seems similar to TestCasePrepared

Reply:
Also, with my retry updates, I was relying on each attempt of a test case getting a new testCaseId (ie testCaseId assigned at testCaseStarted not testCasePrepared). If we want to assign it at testCasePrepared, then that brings us back to a composite key of testCaseId and attempt when trying to record results. 

Thoughts on the following updates:
- TestCasePrepared (sets testCaseId + testStepIds)
- TestCaseStarted (has testCaseId + sets testCaseAttemptId)
- TestStepStarted / Finished (has testCaseAttemptId + testStepId)
- TestCaseFinished (has testCaseAttemptId)